### PR TITLE
fix: Deprecate options.tooltip

### PR DESCRIPTION
**Pull request by elenajdanova**
- Deprecate options.tooltip
- Fixed how we consume tooltip props as we were doing it in 2 different places
- Added additional stories & tests

### CODE REVIEW

**Review by DrewHoo — CHANGES_REQUESTED** [review]
Interweave doesn't look like it is actively maintained & daily downloads are on the low side. Would something like https://www.npmjs.com/package/react-markdown work?

File: src/ui-schema.ts [hunk 1]
```diff
@@ -155,7 +155,7 @@ export type AnyOfControlOptions = OneOfControlOptions
 export type TextControlType = "multiline" | "password" | "singleline"
 
 export type TextControlOptions = {
-  tooltip?: string
+  tooltip?: React.ReactNode
```
Comment (DrewHoo) [inline src/ui-schema.ts:158]: why is this change necessary?

File: src/controls/TextControl.tsx [hunk 2]
```diff
@@ -64,10 +67,10 @@ export function TextControl({
       label={label}
       id={id}
       name={path}
-      rules={rules}
       validateTrigger={["onBlur"]}
-      tooltip={tooltip}
-      {...formItemProps}
+      rules={rules}
+      tooltip={mergedTooltip}
+      {...formItemPropsWOTooltip}
```
Comment (DrewHoo) [inline src/controls/TextControl.tsx:73]: I'm not sure I'm reading this right, but it looks like this code change isn't necessary--does the test you added pass without it?

**Review by DrewHoo — CHANGES_REQUESTED** [review]

File: src/ui-schema.ts (hunk 1 above)
Comment (elenajdanova) [reply src/ui-schema.ts:158]: so we can render HTML here as discussed in [the thread](https://greatexpectationslabs.slack.com/archives/C05FV397N86/p1718743452516369)

Comment (DrewHoo) [reply src/ui-schema.ts:158]: formitem's tooltip is already typed as a ReactNode, though

File: src/controls/TextControl.test.tsx [hunk 3]
```diff
@@ -130,3 +130,33 @@ test("renders error messages from rule validation", async () => {
...
+  const uischema = {
+    type: "VerticalLayout",
+    elements: [
+      {
+        type: "Control",
+        scope: "#/properties/name",
+        label: "Name",
+        options: {
+          tooltip:
+            "Choose <a href='https://wheelofnames.com/' target='_blank' rel='noopener noreferrer'>a random name</a>.",
```
Comment (DrewHoo) [inline src/controls/TextControl.test.tsx:144]: isn't this going to render as a string literal?

File: src/ui-schema.ts (hunk 1 above)
Comment (elenajdanova) [reply src/ui-schema.ts:158]: what is the issue with it?

File: src/controls/TextControl.tsx (hunk 2 above)
Comment (elenajdanova) [reply src/controls/TextControl.tsx:73]: The test will work, but there is some additional ambiguity that needs to be addressed. This is the reason why I dislike having two tooltips. 
If someone passes both tooltips or if we need to handle tooltip properties differently, the `formItem` tooltip will take precedence. Whatever we pass to the tooltip property will be overwritten by the `FormItem` tooltip directly from the ui-schema.

File: src/controls/TextControl.tsx [hunk 4]
```diff
@@ -44,7 +45,9 @@ export function TextControl({
     (uischema.options as TextControlOptions) ?? {}
   const formItemProps =
     "formItemProps" in uischema ? uischema.formItemProps : {}
-  const tooltip = options.tooltip ? options.tooltip : formItemProps?.tooltip
+  const { tooltip, ...formItemPropsWOTooltip } = formItemProps as FormItemProps
```
Comment (DrewHoo) [inline src/controls/TextControl.tsx:48]: Casting here actually makes it possible to create a runtime error if formItemProps is undefined. Instead of casting, you just need to handle the undefined case for formItemProps, e.g. ```const { tooltip, ...formItemPropsWOTooltip } = formItemProps ?? {}```

File: src/controls/TextControl.tsx [hunk 5]
```diff
@@ -44,7 +45,9 @@ export function TextControl({
...
+  const mergedTooltip = options.tooltip ? options.tooltip : tooltip ?? ""
```
Comment (DrewHoo) [inline src/controls/TextControl.tsx:49]: it's not really merged, maybe just call this `tooltip` and alias the tooltip from formItemProps as `formItemTooltip`?

File: src/controls/TextControl.tsx (hunk 2 above)
Comment (DrewHoo) [reply src/controls/TextControl.tsx:73]: Makes sense--maybe add a test that fails without this change, then?

**Review by DrewHoo — CHANGES_REQUESTED** [review]
looking good! couple changes to go, but I like seeing the deprecated annotation in the IDE ❤️

**Review by DrewHoo — APPROVED** [review]
lgtm
